### PR TITLE
Handle the possibility that two workshops will have the same dates bu…

### PR DESCRIPTION
**Pull request by mehalshah**
…t different locations

I took the shorter solution on this - the really right decision would be to use the regional partner workshop IDs instead of searching via location name/address.

### CODE REVIEW

File: dashboard/app/models/pd/application/teacher1819_application.rb [hunk 1]
```diff
@@ -764,7 +764,8 @@ def get_first_selected_workshop
       # Map back to actual workshops by reconstructing the friendly_date_range
       workshops = Pd::Workshop.where(id: workshop_ids)
       hash[:able_to_attend_multiple].each do |response|
-        selected_workshop = workshops.find {|w| response.start_with?(w.friendly_date_range)}
+        location = response.scan(/in (.+) hosted/).first.try(:first) || ''
+        selected_workshop = workshops.find {|w| response.start_with?(w.friendly_date_range) && w.location_address == location}
```
Comment (aoby) [inline dashboard/app/models/pd/application/teacher1819_application.rb]: This should be a 2-pass, using location as a fallback when there are multiple date matches. As we discussed yesterday, many workshop locations were TBD when the applications were submitted, with real locations filled in later. The dates _should_ be more reliable, and haven't changed AFAIK.

Comment (mehalshah) [reply dashboard/app/models/pd/application/teacher1819_application.rb]: That's fair - I'll change it

**Comment by mehalshah** [comment]
That makes sense - I'll do that

File: dashboard/app/models/pd/application/teacher1819_application.rb [hunk 2]
```diff
@@ -764,8 +764,12 @@ def get_first_selected_workshop
...
       workshops = Pd::Workshop.where(id: workshop_ids)
       hash[:able_to_attend_multiple].each do |response|
-        selected_workshop = workshops.find {|w| response.start_with?(w.friendly_date_range)}
-        return selected_workshop if selected_workshop
+        workshops_for_date = workshops.select {|w| response.start_with?(w.friendly_date_range)}
+        return workshops_for_date.first if workshops_for_date.size == 1
+
+        location = response.scan(/in (.+) hosted/).first.try(:first) || ''
+        workshops_for_date_and_location = workshops_for_date.find {|w| w.location_address == location} || workshops_for_date.first
+        return workshops_for_date_and_location if workshops_for_date_and_location
```
Comment (aoby) [inline dashboard/app/models/pd/application/teacher1819_application.rb:784]: nit: we can just return the result of line 771. No need to store it and explicitly return if it exists. 

note: the last lines of a Ruby method being this:
```ruby
value = some_expression
return value if value
```

Presuming, that `value` didn't have a prior value, is always equivalent to:
```ruby
some_expression
```

Comment (mehalshah) [reply dashboard/app/models/pd/application/teacher1819_application.rb:784]: I don't quite understand this - will follow up when you are in

**Pull request merged**